src/set_theory/ordinal.lean

```diff
@@ -1140,6 +1216,27 @@ end
 by rw [limit_rec_on, well_founded.fix_eq,
        dif_neg h.1, dif_neg (not_succ_of_is_limit h)]; refl
 
+lemma has_succ_of_is_limit {α} {r : α → α → Prop} [wo : is_well_order α r]
+  (h : (type r).is_limit) (x : α) : ∃y, r x y :=
```

Comment:
Well, sometimes you want to work with an initial segment of ordinals (mathematically), and do things like taking suprema. Because of universe issues you have to live in a well-order which has as order type some ordinal. Now you start with the assumption that the ordinal is a limit ordinal, and now I want to show that I can take a supremum of successors of elements of `\alpha`. This lemma shows that I can take the successor of an element.

Reply:
Can't you use `enum` to achieve this? That is, you might need the well order type but you can still use ordinals to index them. Or is it too cumbersome?

Reply:
I now rewrote the proof of this to use `enum`. 
It is still two complicated lines to proof this, so I think this lemma has a place in the library, because it would be too cumbersome to do this every time you need this fact.